src/iter/internal.rs

```diff
@@ -166,6 +172,34 @@ impl Splitter {
     }
 }
 
+#[derive(Clone, Copy)]
+struct LengthSplitter {
+    inner: Splitter,
+    min: usize,
+}
+
+impl LengthSplitter {
+    #[inline]
+    fn new(min: usize, max: usize, len: usize) -> LengthSplitter {
+        let mut splitter = LengthSplitter {
+            inner: Splitter::new(),
+            min: cmp::max(min, 1),
+        };
+
+        let max_splits = len / cmp::max(max, 1);
```

Comment:
Perhaps it's poorly named - it represents the number of splits derived from the max, and just below it's only used if it's more than the normal splitter would have used.

I'll fill in a comment later. Thanks for looking it over!